src/netCDF4/__init__.pyi

```diff
@@ -197,80 +296,102 @@ class Dataset:
     def _ncstring_attrs__(self) -> bool: ...
     @property
     def __orthogonal_indexing__(self) -> bool: ...
-
     def filepath(self, encoding: str | None = None) -> str: ...
     def isopen(self) -> bool: ...
     def close(self) -> memoryview: ...  # only if writing and memory != None, but otherwise people ignore the return None anyway
     def sync(self) -> None: ...
     def set_fill_on(self) -> None: ...
     def set_fill_off(self) -> None: ...
-
-    def createDimension(self, dimname: str, size: int | None = None) -> Dimension: ...
-    def renameDimension( self, oldname: str, newname: str) -> None: ...
+    def createDimension(self, dimname: str, size: int | float | None = None) -> Dimension: ...
+    def renameDimension(self, oldname: str, newname: str) -> None: ...
     @overload
-    def createVariable(  # type: ignore
+    def createVariable(
         self,
         varname: str,
-        datatype: T_DatatypeNC,
-        dimensions: DimensionsOptions = (),
-        compression: CompressionOptions | None = None,
-        zlib: bool = False,
-        complevel: CompressionLevelOptions | None = 4,
-        shuffle: bool = True,
-        szip_coding: Literal['nn', 'ec'] = 'nn',
-        szip_pixels_per_block: Literal[4, 8, 16, 32] = 8,
-        blosc_shuffle: Literal[0, 1, 2] = 1,
-        fletcher32: bool = False,
-        contiguous: bool = False,
+        datatype: np.dtype[NumericVarT] | type[NumericVarT],
+        dimensions: DimensionsSpecifier = (),
+        compression: CompressionType | str | None = None,
+        zlib: Any = False,
```

Comment:
Good question. My reasoning was that in practice those parameters allow anything "truthy". I guess the overall principles I eventually came to were to try to avoid false positives as much as possible for people who start using the stubs, even if they're doing something a little weird like using an int or `None` for truthy parameter. This is at odds with type-stubs being a form of self-documentation. I'm open to reverting them back to `bool`.

Reply:
Just because you can doesn't mean you should.
I think bool is the better type hint here, this is widely adopted by other projects.

Reply:
Done in 31513eec.